### ISSUE TITLE
Support multiple files in pathInitScript

### ISSUE THREAD

**Issue by steghio** [alexcojocaru/elasticsearch-maven-plugin#96]
Currently only one file can be specified in pathInitScript.

This can be cumbersome if initialization scripts are already available but not merged in a single file. Please allow for a list of files to be executed instead of a single one.

Bonus points if they can also be executed in the same order they have been listed.

Cheers

**Comment by alexcojocaru** [comment]
That's a reasonable ask. I will take a stab at it in the next few days.

**Comment by steghio** [comment]
I actually have a change I can push for review, but I must do it from home later today. I can put it on a dedicated branch.

One question I have, for backwards compatibility, the configuration property in the POM is called _pathInitScript_ but with this change it would make sense to call it _pathInitScript*s*_

However, in my change I left it the same so as not to break existing usages.

Cheers

**Comment by steghio** [comment]
Hello, turns out I cannot branch and push here. Please get in touch with me otacorp [at] gmail so I can send you what I have if you want.

Cheers

**Comment by alexcojocaru** [comment]
I didn't look too much into this, but one requirement would be backwards compatibility. If you can make the *pathInitScript* property (https://github.com/alexcojocaru/elasticsearch-maven-plugin/blob/master/src/main/java/com/github/alexcojocaru/mojo/elasticsearch/v2/AbstractElasticsearchMojo.java#L121) work both as a string and a string list in the plugin configuration, that would be great. I have a gut feeling that wouldn't work and it has to be either a string, either a string list. That means adding a new property to the config.

**Comment by alexcojocaru** [comment]
> Hello, turns out I cannot branch and push here. Please get in touch with me otacorp [at] gmail so I can send you what I have if you want.
> 
> Cheers

You could fork and start a pull request.

**Comment by alexcojocaru** [comment]
If the forking doesn't work for you for any reason, let me know and I'll get your changes and integrate them manually.

**Comment by steghio** [comment]
Ok nice,  so first some notes and description.

I renamed everywhere to make it easier to track the changes the variable _pathInitScript_ to _pathInitScript**s**_, but the configuration property instead stays the same.

The change is simply to handle in `AbstractElasticsearchMojo` the field `protected String pathInitScripts;` as a comma-separated string of files to run, whatever input we get.

Then the get of that field would return:

`return pathInitScripts.isEmpty() ? new ArrayList<>() : Arrays.asList(pathInitScripts.split(","));`

Thus allowing for no value, single value, multiple values, assuming the string is well formed :)

You can verify this with `PathInitScriptsStringTest` which is not the actual test one should integrate, but it allows for verification that the logic is correct.

And nowhere else is that get used, except for `ClusterConfiguration` where we want it as a List and would therefore have a `List<String>` of files available which are then handled singularily in a loop when needed (`BootstrapClusterStep`, first the verification and then the execution).

I also duplicated the existing `runforked-with-init-script` project as `runforked-with-init-scripts` to test the scenario where more than one file is passed, and added those new files too, I would however think one might want a single test for that.

Unfortunately, I can only run unit tests and cannot run integration tests on my machine because the various IT projects POM files specify this:

`${elasticsearch.maven.plugin.version}`

And I get an error that it is not set anywhere. If I change it to the same as `${project.version}` or even `6.14-SNAPSHOT` I get another error:

`Plugin com.github.alexcojocaru:elasticsearch-maven-plugin:6.14-SNAPSHOT or one of its dependencies could not be resolved: Could not find artifact com.github.alexcojocaru:elasticsearch-maven-plugin:jar:6.14-SNAPSHOT`

I do not have Maven installed on my machine and am using IntelliJ with JDK8.

So long story short, I think this branch https://github.com/steghio/elasticsearch-maven-plugin/tree/feature/96 contains most if not all the changes needed, and I would ask you to please take a look at it and feel free to use what you need to add to your repo.

Of course I am available for clarifications if needed.

Thanks again and have a nice day

**Comment by alexcojocaru** [comment]
Thanks for implementing this, Stefano. I've added a few comments on it, let me know if you have questions. Once they are addressed, feel free to open a PR - the travis build job should run on it and execute the integration tests.

**Comment by steghio** [comment]
Done, PR is opened, just as reminder, I cannot run ITs on my machine, only UTs

Cheers

**Comment by alexcojocaru** [comment]
Thanks for the PR and for addressing all my comments. I have merged it into master and will release a new version of the plugin soon.
